homeassistant/components/climate/xiaomi_miio.py

```diff
@@ -0,0 +1,624 @@
+            self._target_temperature_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+
+        if self._target_temperature < self._target_temperature_low:
+            self._current_operation = 'off'
+            self._target_temperature = self._target_temperature_low
+        elif self._target_temperature > self._target_temperature_high:
+            self._current_operation = 'off'
+            self._target_temperature = self._target_temperature_high
+        elif self._current_temperature and (
+                self._current_operation == "off" or
+                self._current_operation == "idle"):
+            self.set_operation_mode('auto')
+            return
+
+        self.sendcmd()
+        self.schedule_update_ha_state()
+
+    def set_humidity(self, humidity):
+        """Set new target temperature."""
+        self._target_humidity = humidity
+        self.schedule_update_ha_state()
+
+    def set_swing_mode(self, swing_mode):
+        """Set new target temperature."""
+        self._current_swing_mode = swing_mode
+        if self._customize and ('swing' in self._customize):
+            self._customize_sendcmd('swing')
+        else:
+            self.sendcmd()
+        self.schedule_update_ha_state()
+
+    def set_fan_mode(self, fan):
+        """Set new target temperature."""
+        self._current_fan_mode = fan
+        if self._customize and ('fan' in self._customize):
+            self._customize_sendcmd('fan')
+        else:
+            self.sendcmd()
+        self.schedule_update_ha_state()
+
+    def set_operation_mode(self, operation_mode):
+        """Set new target temperature."""
+        self._current_operation = operation_mode
+        self.sendcmd()
+        self.schedule_update_ha_state()
+
+    @property
+    def current_swing_mode(self):
+        """Return the swing setting."""
+        return self._current_swing_mode
+
+    @property
+    def swing_list(self):
+        """List of available swing modes."""
+        return self._swing_list
+
+    def turn_away_mode_on(self):
+        """Turn away mode on."""
+        self._away = True
+        self.schedule_update_ha_state()
+
+    def turn_away_mode_off(self):
+        """Turn away mode off."""
+        self._away = False
+        self.schedule_update_ha_state()
+
+    def set_hold_mode(self, hold):
+        """Update hold mode on."""
+        self._hold = hold
+        self.schedule_update_ha_state()
+
+    def turn_aux_heat_on(self):
+        """Turn away auxillary heater on."""
+        self._aux = True
+        self.schedule_update_ha_state()
+
+    def turn_aux_heat_off(self):
+        """Turn auxillary heater off."""
+        self._aux = False
+        self.schedule_update_ha_state()
+
+    def climate_get_state(self):
+        """get states from climate"""
+        getstate = self.climate.send("get_model_and_state", [])
+        _LOGGER.info(getstate)
+        self._state = ClimateStatus(getstate)
```

Comment:
Here you set the state of an entity do another entity (remember that `ClimateStatus` derives from `ClimateDevice`). That is… weird. `self._state` should be a simple (string) representation of the device's current state.